docs/src/main/asciidoc/native-reference.adoc

```diff
@@ -1304,13 +1304,18 @@ so `gdb` cannot find debugging symbols for our native executable,
 as indicated by the "_No debugging symbols found in ./target/debugging-native-1.0.0-SNAPSHOT-runner_" message in the beginning of `gdb`.
 
 Recompiling the Quarkus application with `-Dquarkus.native.debug.enabled` and rerunning it through `gdb` we are now able to get a backtrace making clear what caused the crash.
-On top of that, add `-H:-OmitInlinedMethodDebugLineInfo` option to avoid inlined methods being omitted from the backtrace:
+On top of that, add `-H:-OmitInlinedMethodDebugLineInfo` option to avoid inlined methods being omitted from the backtrace.
+
+Sources for third party jar dependencies, including Quarkus source code,
+are not added to the source cache by default.
+To include those, make sure you invoke `./mvnw dependency:sources` first, see the basic xref:building-native-image.adoc[Building Native Image Guide].
 
 [source,bash,subs=attributes+]
 ----
 ./mvnw package -DskipTests -Dnative \
     -Dquarkus.native.debug.enabled \
     -Dquarkus.native.additional-build-args=-H:-OmitInlinedMethodDebugLineInfo
+./mvnw dependency:sources
```

Comment:
`dependency:sources` should come first? I would just merge into the command, e.g. `./mvnw dependency:sources package...`

Side note: to be honest, it'd great if adding was already automated by the Quarkus maven plugin. IOW, whenever the user requests getting debug info, add this target automatically. Not sure if that's possible.

For now, I would make sure all debug.enabled examples have it. This guide have 3 such calls.

Also, not sure there's a need to duplicate the explanation. You can point to a section in the original place where debug info is explained and be done with it.

Reply:
> Side note: to be honest, it'd great if adding was already automated by the Quarkus maven plugin. IOW, whenever the user requests getting debug info, add this target automatically. Not sure if that's possible.

It should be possible using something like:

```
<activation>
    <property>
        <name>quarkus.native.debug.enabled</name>
    </property>
</activation>
```

Not sure where it should be placed though (this will need to get generated for Quarkus applications as well not only for the Quarkus core).

Reply:
@zakkak 

>> dependency:sources should come first?
> Yes, otherwise native-image won't copy the java files to the sources directory.

So, the sequence in the test suite has been wrong?
```
mvn package -Pnative -Dquarkus.native.debug.enabled=true 
mvn dependency:sources
```
And then in GDB:
```
set directories /path/to/app/target/quarkus-native-image-source-jar/sources
```

@galderz Thx for pointers, I will amend the text.

Reply:
> @zakkak
> 
> > > dependency:sources should come first?
> > >> Yes, otherwise native-image won't copy the java files to the sources directory.
> 
> So, the sequence in the test suite has been wrong?
> 

It seems so.

> ```
> mvn package -Pnative -Dquarkus.native.debug.enabled=true 
> mvn dependency:sources
> ```
> 
> And then in GDB:
> 
> ```
> set directories /path/to/app/target/quarkus-native-image-source-jar/sources
> ```
> 

If the tests don't check for source code from 3rd party libraries then they won't catch this.

when running `mvn package -Pnative -Dquarkus.native.debug.enabled=true`, `native-image` will create the `sources` directory and copy source files of reachable code there. If you run `mvn dependency:sources` after the fact then the fetched sources won't end up in the `sources` directory as they were not accessible during native image compilation.

Note however that the sources are stored in the `~/.m2/repository` directory so if you are caching this (or when running locally and not on CI) you will have the sources in every run after the first one ;)

Reply:
@zakkak 

> If the tests don't check for source code from 3rd party libraries then they won't catch this.

Hmm, it's in the [application code](https://github.com/Karm/mandrel-integration-tests/blob/master/testsuite/src/it/java/org/graalvm/tests/integration/utils/GDBSession.java#L260) :disappointed: 

So, I'll encode the example from the doc into a TS run to have it covered.

Reply:
> Not sure where it should be placed though (this will need to get generated for Quarkus applications as well not only for the Quarkus core).

This would be something the Quarkus Maven/Gradle plugins would take care of, if feasible. @geoand @gsmet @aloubyansky?

Reply:
> It would make sense to enable it only for native builds, right?

Only for builds that are native and have `quarkus.native.debug.enabled=true`.

Reply:
Feel free to create a separate issue that.

Reply:
Created #31193